src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,114 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private List<Node<K, V>>[] table;
+    private int size;
+    private int threshold;
 
-    @Override
-    public void put(K key, V value) {
+    public MyHashMap() {
+        table = new List[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
 
+    @Override
+    public void put(final K key, V value) {
+        Node<K, V> node = new Node<>(key, value, Objects.hashCode(key));
+        int indexFor = hash(key);
+        int currentSize = size;
+        if (table[indexFor] == null) {
+            initBucket(indexFor);
+            add(indexFor, node);
+        } else {
+            int index = 0;
+            for (Node<K, V> n : table[indexFor]) {
+                if (Objects.equals(node.key, n.key)) {
+                    replace(table[indexFor], n, node, index);
+                    return;
+                }
+                index++;
+            }
+            table[indexFor].add(node);
+            size++;
+        }
+        if (++currentSize > threshold) {
+            resize();
+        }
     }
 
     @Override
-    public V getValue(K key) {
+    public V getValue(final K key) {
+        int indexFor = hash(key);
+        if (table[indexFor] != null) {
+            for (Node<K, V> n : table[indexFor]) {
+                if (Objects.equals(key, n.key)) {
+                    return n.value;
+                }
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static class Node<K, V> {
+        private final int hash;
+        private final K key;
+        private V value;
```

Comment:
Node should contain next Node<K, V>

Reply:
> Node should contain next Node<K, V>

I use an array of lists, in this implementation I don't need next pointer.